gateway-release-common/home/bin/knox-functions.sh

```diff
@@ -250,6 +302,11 @@ function appStop {
      exit 1
    else
      rm -f "$APP_PID_FILE"
+
+     #TODO: due to the current way of shutting down the server the status has never been changed to STOPPED (STARTING seems to be not reliable either which is weird).
```

Comment:
This TODO seems relevant if this isn't being written then this will cause issues. Were you able to figure out why lifecycle isn't writing this file?

Reply:
Currently, the JVM is "stopped" with a `kill` command but there is no [shutdown hook](https://docs.oracle.com/javase/8/docs/api/java/lang/Runtime.html#addShutdownHook-java.lang.Thread-) registered in `GatewayServer`. Adding a shutdown hook would help in most of the cases but it may still send a SIGKILL when stopping a server 'normally' did not succeed.

Reply:
Added the above-mentioned shutdown hook and it's working like a charm.